### PR TITLE
feat(oauth2): logout support

### PR DESCRIPTION
**Pull request by kevintechie**
Adds feature from #612

### CODE REVIEW

File: lib/core/auth.js [hunk 1]
```diff
@@ -153,8 +153,8 @@ export default class Auth {
   }
 
   logout () {
+    this.reset()
```
[the remark this answers is not in the document]
Comment (kevintechie) [reply lib/core/auth.js]: reverted changes

File: lib/schemes/oauth2.js [hunk 2]
```diff
@@ -116,6 +134,18 @@ export default class Oauth2Scheme {
     window.location = url
   }
 
+  logout () {
+    if (!this.options.endpoints.logout) {
```
[the remark this answers is not in the document]
Comment (kevintechie) [reply lib/schemes/oauth2.js]: Made requested changes

**Comment by bramburn** [comment]
nice!

**Comment by pi0** [comment]
Awesome thanks!

**Pull request merged**